games/shift/shift_main.c

```diff
@@ -0,0 +1,927 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <math.h>
+
+#include <nuttx/video/fb.h>
+#include <nuttx/video/rgbcolors.h>
+
+#ifdef CONFIG_GAMES_SHIFT_USE_CONSOLEKEY
+#include "shift_input_console.h"
+#endif
+
+#ifdef CONFIG_GAMES_SHIFT_USE_DJOYSTICK
+#include "shift_input_joystick.h"
+#endif
+
+#ifdef CONFIG_GAMES_SHIFT_USE_GESTURE
+#include "shift_input_gesture.h"
+#endif
+
+/****************************************************************************
+ * Preprocessor Definitions
+ ****************************************************************************/
+
+#define BOARDX_SIZE 6
+#define BOARDY_SIZE 6
+
+#define ROW         8
+#define COL         8
+
+/* Difficult mode:  4, 5, 6 */
+
+#define GAMEMODE    6
+
+#define NCOLORS     GAMEMODE
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct screen_state_s
+{
+  int fd_fb;
+  struct fb_videoinfo_s vinfo;
+  struct fb_planeinfo_s pinfo;
+#ifdef CONFIG_FB_OVERLAY
+  struct fb_overlayinfo_s oinfo;
+#endif
+  FAR void *fbmem;
+};
+
+struct game_screen_s
+{
+  int xres;     /* X display resolution */
+  int yres;     /* Y display resolution */
+  int xoff;     /* X offset to start drawing */
+  int yoff;     /* Y offset to start drawing */
+  int ncolors;  /* Supported number of colors */
+  int blklen;   /* Size of side of the blocks */
+  int steps;    /* Steps to slide a block for a now position */
+  int stepinc;  /* Step increment during the sliding */
+  int dir;      /* Direction to move the blocks */
+};
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const char g_default_fbdev[] = "/dev/fb0";
+
+/* The edge of the board is invisible to user */
+
+int board[ROW][COL] =
+                            {
+                              {1, 1, 1, 1, 1, 1, 1, 1},
+                              {1, 0, 0, 0, 0, 0, 0, 1},
+                              {1, 0, 0, 0, 0, 0, 0, 1},
+                              {1, 0, 0, 0, 0, 0, 0, 1},
+                              {1, 0, 0, 0, 0, 0, 0, 1},
+                              {1, 0, 0, 0, 0, 0, 0, 1},
+                              {1, 0, 0, 0, 0, 0, 0, 1},
+                              {1, 1, 1, 1, 1, 1, 1, 1}
+                            };
```

Comment:
please remove extra spaces

Reply:
Ok, my plan was to keep it aligned with the end of previous line, but I will move it to the left